File: scratch/find_q11_html.php
<?php

$htmlContent = file_get_contents('C:\Users\LENOVO\Downloads\AZ-303-220-sequential-clean-final.html');

libxml_use_internal_errors(true);
$dom = new \DOMDocument();
$dom->loadHTML(mb_convert_encoding($htmlContent, 'HTML-ENTITIES', 'UTF-8'));
libxml_clear_errors();

$xpath = new \DOMXPath($dom);
$q11Nodes = $xpath->query('//article[@id="q11"]');

if ($q11Nodes->length > 0) {
    echo $dom->saveHTML($q11Nodes->item(0));
} else {
    echo "q11 not found";
}

File: scratch/find_q11_html.py
from bs4 import BeautifulSoup

html_path = r"C:\Users\LENOVO\Downloads\AZ-303-220-sequential-clean-final.html"

with open(html_path, 'r', encoding='utf-8') as f:
    content = f.read()

soup = BeautifulSoup(content, 'html.parser')
q11 = soup.find('article', id='q11')

if q11:
    print("--- Q11 HTML FROM MASTER FILE ---")
    print(q11.prettify()[:3000])
else:
    print("q11 not found in HTML!")
